### PR TITLE
refactored Omnibox to be more declarative

### PR DESCRIPTION
**Pull request by PoisonousJohn**
Well I wanted to focus Omnibox when I open empty tab, but I understood that Omnibox's logic was too complex.

So I tried to refactor it using Declarative State Machine, and it seems more understandable to me now.

What do you think?

### CODE REVIEW

**Comment by PoisonousJohn** [comment]
Strange Travis error

**Comment by timsueberkrueb** [comment]
Hey, cool, I'm will try to check it out later today or tomorrow evening!

**Comment by timsueberkrueb** [comment]
What is the advantage of using `StateMachine` over the ["normal" states](http://doc.qt.io/qt-5/qml-qtquick-state.html) here? (e.g. wouldn't `states: [ State { ... }, State {...}, State {...}]` be sufficient?).

**Comment by timsueberkrueb** [comment]
I see, `SignalTransition`s *are* handy here, nice.

File: src/ui/omnibox/Omnibox.qml [hunk 1]
```diff
@@ -170,16 +165,64 @@ Item {
...
+        initialState: idle
+        running: true
+
+        DSM.State {
+            id: idle
+
+            onEntered: {
+                checkForEmptyTab();
+                textField.text = showUrlField.text;
+            }
```
Comment (timsueberkrueb) [inline src/ui/omnibox/Omnibox.qml:185]: Nitpick: It would be nice if you could insert an empty line here.

File: src/ui/omnibox/Omnibox.qml [hunk 2]
```diff
@@ -170,16 +165,64 @@ Item {
...
+            DSM.SignalTransition {
+                signal: showUrlField.activeFocusChanged
+                guard: showUrlField.activeFocus === true
+                targetState: editing
+            }
```
Comment (tim-sueberkrueb) [inline src/ui/omnibox/Omnibox.qml]: Empty line here.

**Review by timsueberkrueb — APPROVED** [review]
Looking good to me

**Comment by timsueberkrueb** [comment]
@plfiorini can you have a quick look at this as well? Do you agree?

**Review by plfiorini — APPROVED** [review]

**Pull request merged**